### PR TITLE
virt-operator: remove Kubevirt service accounts from default privileged SCC upon upgrade

### PR DESCRIPTION
**Pull request by enp0s3**
During an upgrade the virt-operator will delete Kubevirt service accounts from the 
default Security Context Constraint called "privileged". That will occur if the 
cluster provider is Openshift.

 Signed-off-by: Igor Bezukh <ibezukh@redhat.com>

**What this PR does / why we need it**:
When running in Openshift we create our own Security Context Constraints and add our service accounts there.
No need to use Openshift default SCCs such as the privileged SCC which was used in prior Kubevirt releases.
Moreover, it can introduce a security issue since the default privileged SCC may have more relaxed policy 
than needed.

**Which issue(s) this PR fixes** 
https://bugzilla.redhat.com/show_bug.cgi?id=1860992

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-operator/install-strategy/create.go
Comment (davidvossel) [inline pkg/virt-operator/install-strategy/create.go:2113]: Your logic is great. Based on the context of what has changed regarding SCC's I'm going to suggest a slightly different approach.

This Add/Remove SCC user logic was the only reason we needed the prevStrategy that I'm aware of. I don't ever anticipate us wanting this logic to alter existing SCCs again, so I think we should use this as an opportunity to get rid of it entirely.

The change I'm proposing would then be to
- remove usage of prevStrategy from both create.go and kubevirt.go
- remove the addOrRemoveSCC function
- add a new function that simply unconditionally removes our SAs from the privileged SCC if they are set (regardless of what is in the install strategy)

**Comment by davidvossel** [comment]
needs rebase

File: api/openapi-spec/swagger.json
Comment (enp0s3) [reply api/openapi-spec/swagger.json]: @davidvossel Done. I've redesigned the PR as per your suggestion.

File: pkg/virt-operator/install-strategy/create_test.go
Comment (danielBelenky) [inline pkg/virt-operator/install-strategy/create_test.go]: Maybe you should use a `DescribeTable` here, instead of duplicating the tests.

File: pkg/virt-operator/creation/rbac/operator.go
Comment (danielBelenky) [inline pkg/virt-operator/creation/rbac/operator.go:498]: What about the operator's SA?

File: pkg/virt-operator/creation/rbac/operator.go
Comment (danielBelenky) [inline pkg/virt-operator/creation/rbac/operator.go:500]: looks like a redundant change... I think that the problem is that in your previous commit you changed this without a reason...

File: pkg/virt-operator/install-strategy/create.go
Comment (danielBelenky) [inline pkg/virt-operator/install-strategy/create.go:3277]: You should check for the error first to be on the safe side. Otherwise, if somehow `exists` will be true and `err != nil`, you will never get to the error block.

File: pkg/virt-operator/install-strategy/create.go
Comment (danielBelenky) [inline pkg/virt-operator/install-strategy/create.go:3287]: This copy is redundant IMO. You can just initialize an empty `users` list and just modify it whenver you remove a user.

File: pkg/virt-operator/install-strategy/create.go
Comment (danielBelenky) [inline pkg/virt-operator/install-strategy/create.go:3292]: If you will create a map from the users, you will reduce the runtime of this loop from O(n^2) to O(n)

File: pkg/virt-operator/install-strategy/create.go
Comment (danielBelenky) [inline pkg/virt-operator/install-strategy/create.go:3272]: Just a nit: I'd prefer to set this variable closer to the place where you're going to use it. Just to make it easier to find.

File: pkg/virt-operator/install-strategy/create_test.go
Comment (danielBelenky) [inline pkg/virt-operator/install-strategy/create_test.go:785]: Here you could probably use a `DescribeTable` too...

File: pkg/virt-operator/kubevirt.go
Comment (danielBelenky) [inline pkg/virt-operator/kubevirt.go:838]: Since you removed it, `loadObservedVersion` bool can be removed from the function's signature.

**Review by danielBelenky — CHANGES_REQUESTED** [review]

File: pkg/virt-operator/kubevirt.go
Comment (davidvossel) [reply pkg/virt-operator/kubevirt.go:838]: +1 good catch

File: pkg/virt-operator/install-strategy/create.go
Comment (davidvossel) [reply pkg/virt-operator/install-strategy/create.go:3292]: yeah, so if you had a map of the kubevirt users, it would just be a direct lookup.

```
if ok := kvSericeAccounts[acc]; ok {


}
```

**Review by davidvossel — COMMENTED** [review]
/approve

nice refactor! looks great. There are just a few comments to address. I went ahead and gave this an approve so all you'll need is a lgtm once the revisions are complete.

**Comment by davidvossel** [comment]
/approved

umm tag wasn't added last time. i'm retrying

File: pkg/virt-operator/install-strategy/create_test.go
Comment (enp0s3) [reply pkg/virt-operator/install-strategy/create_test.go]: @danielBelenky Thank you. Done.

File: pkg/virt-operator/creation/rbac/operator.go
Comment (enp0s3) [reply pkg/virt-operator/creation/rbac/operator.go]: @danielBelenky Thank you. Done.

File: pkg/virt-operator/install-strategy/create.go
Comment (enp0s3) [reply pkg/virt-operator/install-strategy/create.go]: @danielBelenky Thank you. Done.

File: pkg/virt-operator/install-strategy/create.go
Comment (enp0s3) [reply pkg/virt-operator/install-strategy/create.go]: @danielBelenky Thank you. Done.

File: pkg/virt-operator/install-strategy/create.go
Comment (enp0s3) [reply pkg/virt-operator/install-strategy/create.go]: @danielBelenky @davidvossel Thank you. Done.

File: pkg/virt-operator/install-strategy/create.go
Comment (enp0s3) [reply pkg/virt-operator/install-strategy/create.go]: @danielBelenky Thank you. Done.

File: pkg/virt-operator/install-strategy/create_test.go
Comment (enp0s3) [reply pkg/virt-operator/install-strategy/create_test.go:804]: @danielBelenky Thank you. Done.

File: pkg/virt-operator/kubevirt.go
Comment (enp0s3) [reply pkg/virt-operator/kubevirt.go:838]: @danielBelenky @davidvossel Thank you. Done.

**Review by davidvossel — APPROVED** [review]
/lgtm

**Review by xpivarc — APPROVED** [review]
/lgtm

**Pull request merged**
